libs/mpi/CMakeLists.txt

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2019-2020 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+# if networking is off, or MPI parcelport is not being used
+# then allow the option of using our asynchronous MPI features
+if(HPX_WITH_NETWORKING AND HPX_WITH_PARCELPORT_MPI)
+  hpx_option(HPX_MPI_WITH_FUTURES BOOL
+    "Enable support for returning futures from MPI asynchronous calls"
+    " (default: ON if networking+MPI already enabled)"
+    ON CATEGORY "MPI")
+else()
+  hpx_option(HPX_MPI_WITH_FUTURES BOOL
+    "Enable support for returning futures from MPI asynchronous calls"
+    " (default: ON if networking+MPI already enabled)"
+    OFF CATEGORY "MPI")
+endif()
+
+# if the user does not want mpi async futures, quit - the module will not be enabled
+if (NOT ${HPX_MPI_WITH_FUTURES})
+  return()
+endif()
+
+# setup MPI, if necessary
+if(NOT TARGET Mpi::mpi)
+  include(HPX_SetupMPI)
+  setup_mpi()
+endif()
+
+# Default location is $HPX_ROOT/libs/mpi/include
+set(mpi_headers
+    hpx/mpi/force_linking.hpp
```

Comment:
Same here, I wonder if we can't just use the usual force linking functionality.

Reply:
I'll create a separate ticket for this.